### PR TITLE
PC: shellcheck log_instance.sh restart_instance.sh

### PR DESCRIPTION
**Pull request by pdostal**
- Related pull request: #17250
- Verification run: [pdostal-server.suse.cz/t4471](https://pdostal-server.suse.cz/tests/4471)

### CODE REVIEW

**Review by grisu48 — APPROVED** [review]

File: data/publiccloud/log_instance.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ ec2_stop_log()
     aws ec2 get-console-output --instance-id "$INSTANCE_ID" --output text > "${OUTPUT_DIR}/${CNT}_get_console_output_stop.log"
     set -e
     if [ -f "$PID_FILE" ]; then
-      kill -9 $(cat "$PID_FILE")
+      kill -9 "$(cat "$PID_FILE")"
```
Comment (ricardobranco777) [inline data/publiccloud/log_instance.sh:33]:
```suggestion
      kill -9 $(< "$PID_FILE")
```

File: data/publiccloud/log_instance.sh [hunk 2]
```diff
@@ -89,7 +90,7 @@ gce_stop_log()
     # Use flock to wait max 60s that gce_read_serial() is ready. And kill during
     # the 30s sleep afterwards. If flock fail, just kill the bg process.
     (flock -w 60 -e 9 || exit 1
-        gce_is_running && kill -9 $( cat "$PID_FILE" )
+        gce_is_running && kill -9 "$( cat "$PID_FILE" )"
```
Comment (ricardobranco777) [inline data/publiccloud/log_instance.sh:93]:
```suggestion
        gce_is_running && kill -9 $(< "$PID_FILE")
```

File: data/publiccloud/log_instance.sh [hunk 3]
```diff
@@ -19,7 +19,7 @@ ec2_start_log()
     set +e
     aws ec2 get-console-output --instance-id "$INSTANCE_ID" --output text > "${OUTPUT_DIR}/${CNT}_get_console_output_start.log"
     set -e
-    nohup ssh $SSH_OPTS "ec2-user@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 &
+    nohup ssh "$SSH_OPTS" "ec2-user@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 &
```
Comment (ricardobranco777) [inline data/publiccloud/log_instance.sh:22]:
```suggestion
    nohup ssh $SSH_OPTS "ec2-user@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 & # shellcheck disable=SC2086
```

We have this variable defined above as:

`SSH_OPTS="-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR"`

File: data/publiccloud/log_instance.sh [hunk 4]
```diff
@@ -101,7 +102,7 @@ azure_start_log()
     set +e
     az vm boot-diagnostics get-boot-log --ids "$INSTANCE_ID" > "${OUTPUT_DIR}/$CNT""_boot_log_start.txt" 2>&1
     set -e
-    nohup ssh $SSH_OPTS "azureuser@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 &
+    nohup ssh "$SSH_OPTS" "azureuser@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 &
```
Comment (ricardobranco777) [inline data/publiccloud/log_instance.sh:105]:
```suggestion
    nohup ssh $SSH_OPTS "azureuser@${HOST}" -- sudo dmesg -c -w > "${OUTPUT_DIR}/${CNT}_dmesg.log" 2>&1 & # shellcheck disable=SC2086
```

File: data/publiccloud/log_instance.sh [hunk 5]
```diff
@@ -116,7 +117,7 @@ azure_stop_log()
     az vm boot-diagnostics get-boot-log --ids "$INSTANCE_ID" > "${OUTPUT_DIR}/$CNT""_boot_log_stop.txt" 2>&1
     set -e
     if [ -f "$PID_FILE" ]; then
-      kill -9 $(cat "$PID_FILE")
+      kill -9 "$(cat "$PID_FILE")"
```
Comment (ricardobranco777) [inline data/publiccloud/log_instance.sh:120]:
```suggestion
      kill -9 $(< "$PID_FILE")
```

**Review by ricardobranco777 — CHANGES_REQUESTED** [review]
The VR will fail as it is because `$SSH_OPTS` shouldn't be quoted.  Also added some suggestions to avoid using `cat`.

**Comment by pdostal** [comment]
image

**Review by ricardobranco777 — APPROVED** [review]
LGTM

File: data/publiccloud/log_instance.sh [hunk 6]
```diff
@@ -19,6 +19,7 @@ ec2_start_log()
     set +e
     aws ec2 get-console-output --instance-id "$INSTANCE_ID" --output text > "${OUTPUT_DIR}/${CNT}_get_console_output_start.log"
     set -e
+    # shellcheck disable=2086
```
Comment (asmorodskyi) [inline data/publiccloud/log_instance.sh:22]: what does it mean ?

Comment (asmorodskyi) [reply data/publiccloud/log_instance.sh:22]: ok now I see

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by asmorodskyi** [comment]
IMHO it make more sense to have this changes together with https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17250  ( PR which introducing pipeline also tweak files which not passing it ) but it is not super critical

**Pull request merged**